crypto/fipsmodule/ec/ec.c

```diff
@@ -506,17 +492,22 @@ int EC_GROUP_cmp(const EC_GROUP *a, const EC_GROUP *b, BN_CTX *ignored) {
     return 0;
   }
 
-  // |a| and |b| are both custom curves. We compare the entire curve
-  // structure. If |a| or |b| is incomplete (due to legacy OpenSSL mistakes,
-  // custom curve construction is sadly done in two parts) but otherwise not the
-  // same object, we consider them always unequal.
-  return a->meth != b->meth ||  //
-         !a->has_order || !b->has_order ||
-         BN_cmp(&a->order.N, &b->order.N) != 0 ||
-         BN_cmp(&a->field.N, &b->field.N) != 0 ||
-         !ec_felem_equal(a, &a->a, &b->a) ||  //
-         !ec_felem_equal(a, &a->b, &b->b) ||
-         !ec_GFp_simple_points_equal(a, &a->generator.raw, &b->generator.raw);
+  // |a| and |b| are both custom curves. We compare the entire curve structure
+  // if both |a| and |b| are complete. If either are incomplete (due to legacy
```

Comment:
I think this needs to be updated now
```
If either are incomplete ... we only compare the parts that are available.
```

Reply:
Ah yes, will change to "if both are incomplete"